program/src/initialize_vault_operator_delegation_snapshot.rs

```diff
@@ -0,0 +1,156 @@
+use jito_bytemuck::AccountDeserialize;
+use jito_restaking_core::{
+    config::Config, ncn::Ncn, ncn_vault_ticket::NcnVaultTicket, operator::Operator,
+};
+use jito_tip_router_core::{
+    epoch_snapshot::{EpochSnapshot, OperatorSnapshot},
+    loaders::load_ncn_epoch,
+    ncn_config::NcnConfig,
+    weight_table::WeightTable,
+};
+use jito_vault_core::{
+    vault::Vault, vault_ncn_ticket::VaultNcnTicket,
+    vault_operator_delegation::VaultOperatorDelegation,
+};
+use solana_program::{
+    account_info::AccountInfo, clock::Clock, entrypoint::ProgramResult, msg,
+    program_error::ProgramError, pubkey::Pubkey, sysvar::Sysvar,
+};
+
+pub fn process_initialize_vault_operator_delegation_snapshot(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    first_slot_of_ncn_epoch: Option<u64>,
+) -> ProgramResult {
+    let [ncn_config, restaking_config, ncn, operator, vault, vault_ncn_ticket, ncn_vault_ticket, vault_operator_delegation, weight_table, epoch_snapshot, operator_snapshot, vault_program, restaking_program] =
+        accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    if vault_program.key.ne(&jito_vault_program::id()) {
+        msg!("Incorrect vault program ID");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    if restaking_program.key.ne(&jito_restaking_program::id()) {
+        msg!("Incorrect restaking program ID");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    NcnConfig::load(program_id, ncn.key, ncn_config, false)?;
+    Config::load(restaking_program.key, restaking_config, false)?;
+    Ncn::load(restaking_program.key, ncn, false)?;
+    Operator::load(restaking_program.key, operator, false)?;
+    Vault::load(vault_program.key, vault, false)?;
+
+    VaultOperatorDelegation::load(
+        vault_program.key,
+        vault_operator_delegation,
+        vault,
+        operator,
+        false,
+    )?;
+    VaultNcnTicket::load(vault_program.key, vault_ncn_ticket, vault, ncn, false)?;
+    NcnVaultTicket::load(restaking_program.key, ncn_vault_ticket, ncn, vault, false)?;
+
+    //TODO check that st mint is supported?
+    //TODO may not exist
+    //TODO what happens if a new Vault is added inbetween weight table creation and this? - There could be a count mismatch
```

Comment:
Could we just use the count from weight table to avoid this issue?

Reply:
yep